packages/flutter/lib/src/material/input_decorator.dart

```diff
@@ -2430,7 +2443,12 @@ class InputDecoration {
     this.errorText,
     this.errorStyle,
     this.errorMaxLines,
+    @Deprecated(
+      'Use floatLabelBehaviour instead. '
+      'This feature was deprecated after v1.13.2.'
+    )
     this.hasFloatingPlaceholder = true,
+    this.floatLabelBehavior = FloatLabelBehavior.auto,
```

Comment:
Does it make sense to allow certain strange combinations like `!hasFloatingPlaceholder && floatLabelBehavior == FloatLabelBehavior.always`?  You could add assertions at the bottom of these constructors to prevent cases like that if it would make things simpler.  It's a bit complicated given that hasFloatingPlaceholder is deprecated though.

Reply:
Note that if you do add assertions, it will probably affect your tests.

Reply:
yeah actually I think an assertion that asserts that only one of hasFloatingPlaceholder or floatLabelBehavior is used would be good.

Reply:
although that would mean no default values, so perhaps an assertion on specific combinations is better